robocluster/device.py

```diff
@@ -111,6 +114,32 @@ def sleep(self, duration):
             for callback in self.events[event]:
                 self._loop.create_task(callback(event, data))
 
+    async def _serial_read_task(self):
+        """
+        Recieve packets from a linked serial device
+        and call the appropriate callbacks.
+        """
+        while not self._serial_device.isInitialized():
+            await self.sleep(0.01)  # Better way to do this?
+        while True:
+            packet = await self._serial_device.read_packet()
+            event, data = packet['event'], packet['data']
+            for callback in self.events[event]:
+                self._loop.create_task(callback(event, data))
+
+    def link_serial(self, serialdevice):
```

Comment:
Would we ever want to have a simple "SerialDevice" that would just publish directly from the serial reads?

Reply:
For things such as logging, this may be usefull. I also want to separate the processes that talk directly to hardware and processes that make logical decisions, as you can test the logical layer easily by generating simulated values without changing anything. https://github.com/UofSSpaceTeam/usst-docs/wiki/Software-System-Overview So the processes that talk to the hardware devices just translate between robocluster and whatever binary protocol the hardware talks in. If that happens to be json, it can just be a simple passthrough device.
We also don't have to just publish the data. Another use case would be to have two microprocessors talk to each other "directly" and the Device just connects the two streams.